### PR TITLE
cache: remove cache space constraint

### LINKED ISSUE #6603: Is there any way to change the limit on the cache object size?
I have realized that the maximum size of the objects in the minio cache is a function of the free space on the minio cache disk - only objects a 100 times smaller that the remaining free space get cached.
Is there any way to tweak / circumvent this constraint?

## Expected Behavior
Minio should cache objects of any sizes

## Current Behavior
Minio only caches objects 100 times smaller that the remaining free space on the minio cache disk

## Possible Solution
Maybe add an environment variable allowing the tweaking of this setting

## Steps to Reproduce (for bugs)
This is clearly not a bug, just not flexible for all use-cases.

## Context
We are transferring quite big files (3- 4 TB) so we have a hard time to finf the appropriate storage for minio cache disks

## Your Environment
Centos 7.4
Minio latest - mode: minio azure gateway
XEN virtual machine

### PR DESCRIPTION
**Pull request by poornas**
relax cache constraint of requiring 100 times size of object
being cached for better cache utilization.

## Motivation and Context

Fixes #6603. Currently, disk cache has a heuristic in place to require 100 times size of object being uploaded to object storage. Relax this constraint,as it will facilitate better cache utilization.
## Regression
 No

## How Has This Been Tested?

Run minio in fs mode with cache enabled, i.e. set MINIO_CACHE_DRIVES and tweak the MINIO_CACHE_MAXUSE % based on available disk space on your system to test the edge case of cache space running out for a large enough upload.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/disk-cache.go [hunk 1]
```diff
@@ -231,7 +230,7 @@ func (c cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string,
 		// We don't cache partial objects.
 		return c.GetObjectNInfoFn(ctx, bucket, object, rs, h, writeLock, opts)
 	}
-	if !dcache.diskAvailable(objInfo.Size * cacheSizeMultiplier) {
+	if !dcache.diskAvailable(objInfo.Size) {
 		// cache only objects < 1/100th of disk capacity
```
Comment (kannappanr) [inline cmd/disk-cache.go]: The comment is invalid now.

File: cmd/disk-cache.go [hunk 2]
```diff
@@ -308,7 +307,7 @@ func (c cacheObjects) GetObject(ctx context.Context, bucket, object string, star
 		// We don't cache partial objects.
 		return GetObjectFn(ctx, bucket, object, startOffset, length, writer, etag, opts)
 	}
-	if !dcache.diskAvailable(objInfo.Size * cacheSizeMultiplier) {
+	if !dcache.diskAvailable(objInfo.Size) {
 		// cache only objects < 1/100th of disk capacity
```
Comment (kannappanr) [inline cmd/disk-cache.go]: Please remove or reword this comment

Comment (poornas) [reply cmd/disk-cache.go]: done

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]
